titiler/errors.py

```diff
@@ -33,8 +43,10 @@ class BadRequestError(TilerError):
     TileOutsideBounds: status.HTTP_404_NOT_FOUND,
     TileNotFoundError: status.HTTP_404_NOT_FOUND,
     MosaicNotFoundError: status.HTTP_404_NOT_FOUND,
+    NoAssetFoundError: status.HTTP_404_NOT_FOUND,
     RasterioIOError: status.HTTP_404_NOT_FOUND,
-    InvalidBandName: status.HTTP_404_NOT_FOUND,
+    MissingAssets: status.HTTP_404_NOT_FOUND,
```

Comment:
should this return `status.HTTP_400_BAD_REQUEST` @geospatial-jeff ?

Reply:
I think 400 is appropriate here since this exception will be raised if the caller doesn't provide assets/expression.